src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,78 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.exceptions.MissedRequiredAnnotationException;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
+    private Map<Class<?>, Class<?>> interfaceImplementations;
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (interfaceClazz.isAnnotationPresent(Component.class)) {
+            Object classImplInstance = null;
```

Comment:
flip the `if` to follow the return/fail early pattern
```suggestion
        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
                throw new MissedRequiredAnnotationException("To create instance we need "
                + "to have Component annotation");
        }
        Object classImplInstance = null;
        // other logic
```